### DISCUSSION TITLE
async onMount

### DISCUSSION THREAD

**Discussion by AjaxSolutions** [pmndrs/jotai#1227] [General] [answered]
Can the onMount function be async?

**Comment by dai-shi** [comment, marked as answer]
Do you mean returning cleanup function in async? I don't think it's possible like `useEffect` can't.

A possible workaround:
```js
anAtom.onMount = () => {
  let cleanup;
  (async () => {
    // ...
    cleanup = ...
  })();
  return () => cleanup?.();
};
```
But, as you see, this is not safe. You should check if it's already unmounted. I think we use that technique in: https://github.com/pmndrs/jotai/blob/492544e1845bf2f6d4152242bc64f8dd7d0ccdc0/src/xstate/atomWithMachine.ts#L97-L112
